solr/core/src/test/org/apache/solr/parser/SolrQueryParserBaseTest.java

```diff
@@ -26,19 +24,17 @@
 import java.util.List;
 import org.apache.lucene.queryparser.charstream.CharStream;
 import org.apache.lucene.search.Query;
+import org.apache.solr.SolrTestCaseJ4;
 import org.apache.solr.common.SolrException;
 import org.apache.solr.request.SolrQueryRequest;
 import org.apache.solr.schema.IndexSchema;
 import org.apache.solr.search.QParser;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.Mockito;
 
-@RunWith(MockitoJUnitRunner.class)
-public class SolrQueryParserBaseTest {
+public class SolrQueryParserBaseTest extends SolrTestCaseJ4 {
```

Comment:
Get all the benefits of the underlying randomization and thread leak checks from Lucene. This ensures that all these tests have a common base. There are no real downsides to doing this either.

This is still using mocks just instead of the MockitoJUnitRunner uses the Lucene runner. This also matches all other Mockito uses in the Solr project - this is the ONLY class using MockitoJUnitRunner